### PR TITLE
Java: Convert remote flow sources to use new CSV format

### PR DESCRIPTION
**Pull request by tamasvajk**
This PR converts some of the remote flow sources to CSV based ones.

~~[Differences job](https://jenkins.internal.semmle.com/job/Changes/job/Java-Differences/1195/)~~ [Differences job](https://jenkins.internal.semmle.com/job/Changes/job/Java-Differences/1211/)

### CODE REVIEW

**Comment by intrigus-lgtm** [comment]
What is the advantage of using the CSV format instead of using the old modelling?
IMHO explicit modelling may be more verbose, but also easier to understand than a very large CSV file.
I personally would prefer explicit modelling.

I see that CSV might be useful when a customer wants to define custom sources and can not edit the QL files.

**Comment by tamasvajk** [comment]
> What is the advantage of using the CSV format instead of using the old modelling?
> IMHO explicit modelling may be more verbose, but also easier to understand than a very large CSV file.
> I personally would prefer explicit modelling.
> 
> I see that CSV might be useful when a customer wants to define custom sources and can not edit the QL files.

There's no technical benefit of one over the other. We want to see the limitations of the current CSV format, hence the migration. Also, the more cases we cover with the CSV format, the more understandable it becomes, and we think it will be easier for everyone to add new APIs to the CSV format.

File: java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll [hunk 1]
```diff
@@ -86,7 +86,82 @@ private predicate sourceModelCsv(string row) {
...
+      "org.springframework.web.multipart;MultipartRequest;true;getMultipartContentType;(String);;ReturnValue;remote",
+      // SpringMultipartFileSource
+      "org.springframework.web.multipart;MultipartFile;true;getBytes;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getContentType;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getInputStream;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getName;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getOriginalFilename;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getResource;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;getSize;();;ReturnValue;remote",
+      "org.springframework.web.multipart;MultipartFile;true;isEmpty;();;ReturnValue;remote",
```
Comment (aschackmull) [inline java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll:105]: This line looks added by mistake. I don't think `isEmpty` should be included.

**Comment by aschackmull** [comment]
One minor comment, otherwise LGTM. We should probably get rid of some of the now likely unused library classes, but let's save that for a followup PR.

File: java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll (hunk 1 above)
Comment (tamasvajk) [reply java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll:105]: Previously only `get%` methods were covered, but now that I had to add the methods one by one, I also added `isEmpty`. Don't you think it's analogous to `getSize`?

Comment (aschackmull) [reply java/ql/src/semmle/code/java/dataflow/ExternalFlow.qll:105]: Somewhat, yes. Let's remove `getSize` as well. I'd say that a boolean can't be tainted (it's simply too few bits). A long can be considered tainted in some contexts, but this is mostly relevant for C/C++ and not so often for Java. And regardless I don't think measuring the size of some tainted data should be considered tainted.

**Comment by tamasvajk** [comment]
@aschackmull I rebased this PR to fix the merge conflicts. Also, I pushed a couple of extra commits to handle `java.net.http.WebSocket$Listener.onText`, the new `play.mvc.Http$RequestHeader.*` and to take your feedback into account.

I'm starting a new diff job...

**Review by aschackmull — COMMENTED** [review]
LGTM.

**Review by aschackmull — APPROVED** [review]

**Pull request merged**